extension/ballastextension/factory.go

```diff
@@ -20,13 +20,17 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/extension/extensionhelper"
+	"go.opentelemetry.io/collector/internal/iruntime"
 )
 
 const (
 	// The value of extension "type" in configuration.
 	typeStr = "memory_ballast"
 )
 
+// memHandler returns the total memory of the target host/vm
+var memHandler = iruntime.TotalMemory
```

Comment:
You don't need this global. Tests can call newMemoryBallast with their handlers and createExtension can always pass iruntime.TotalMemory.

Reply:
I guess this is required for a test below. If not, can we remove it?

Reply:
yes. it is required for the tests.